pymodbus/framer/rtu_framer.py

```diff
@@ -60,7 +60,7 @@ def __init__(self, decoder, client=None):
         :param decoder: The decoder factory implementation to use
         """
         self._buffer = b''
-        self._header = {'uid': 0x00, 'len': 0, 'crc': '0000'}
+        self._header = {}
```

Comment:
It looks like `isFrameReady()` can still work incorrectly in some conditions.

If framer instance receives incomplete frame with only first byte of some frame, 
`populateHeader()` properly sets `self._header['uid']` and then raises `IndexError` when trying to read second byte from buffer. After that `isFrameReady()` will not call `populateHeader` because `self._header` is not empty, but instead will fail on line `return self._header and len(self._buffer) >= self._header['len']`, as `len` key was not added to `self._header`